server/code_switch/code_switch_unit.py

```diff
@@ -1,134 +1,92 @@
-        self.current_cs_state = self.cs_history[-1]
-
-
-    def __generate_response(self, en_bot_resp: List[str]) -> List[str]:
-        spanglish_bot_response_list = []
-        for eng_resp in en_bot_resp:
-            new_cs_state = self.__predict_next_cs_state()
-            self.__update_cs_state(new_cs_state)
-            spanglish_bot_response = self.translation[new_cs_state](eng_resp)
-            spanglish_bot_response_list.append(spanglish_bot_response)
-        return spanglish_bot_response_list
-
-    def __identify_incoming_cs_state(self):
-        self.current_cs_state = langid.classify(self.user_msg)[0]
-
-    def __predict_next_cs_state(self):
-        if self.current_cs_state is None:  # not-initialized
-            return self.__random_strategy()
-
-        return self.strategy[self.cs_strategy]()
-
-    def __update_cs_state(self, new_state):
-        self.current_cs_state = new_state
-        self.cs_history.append(new_state)
-
-    def __goldfish_cs_strategy(self):
-        r = self.params[self.current_cs_state].r
-        p = hazard(r, s=self.len_of_current_subsequence)
-        res = random.choices([True, False], [p, 1 - p])[0]
-        if res:  # change state
-            next_state = self.__select_next_state(self.current_cs_state)
-            self.len_of_current_subsequence = 1
-        else:  # keep current state
-            next_state = self.current_cs_state
-            self.len_of_current_subsequence += 1
-        return next_state
-
-    def __select_next_state(self, current_state):
-        all_next_possible_state_options = []
-        all_next_possible_state_probabilities = []
-        for next_state_option, next_state_transition_probability in \
-                self.params[current_state].transitions.items():
-
-            if not(next_state_option == current_state):
-                all_next_possible_state_options.append(next_state_option)
-                all_next_possible_state_probabilities.append(next_state_transition_probability)
-        next_state = random.choices(all_next_possible_state_options, all_next_possible_state_probabilities)[0]
-        return next_state
-
-    def __random_strategy(self):
-        weights = []
-        for lang in self.params:
-            cs_option: CSOption = self.params[lang]
-            weights.append(cs_option.probability)
-        return random.choices(list(self.params.keys()), weights)[0]
-
-    def __tit_for_tat_strategy(self):
-        next_state = self.default_lang
-        previous_user_lang = langid.classify(self.user_msg)[0]
-        if previous_user_lang is not None:
-            next_state = previous_user_lang
-        return next_state
-
-    @staticmethod
-    def __english_only_strategy():
-        return 'en'
-
-    @staticmethod
-    def __spanish_only_strategy():
-        return 'es'
-
+PUNCTUATIONS_MARKS = ['!', '?', ',', ':', ';', '.']
 
+class CodeSwitchUnit:
+	def __init__(self, cs_strategy: str):
+		self.params = DEFAULT_CS_PARAMS
```

Comment:
should the main code switching unit know this param- the cs params? isn't it enough that the cs_strategy super class knows this? if its avilable to the super class, all children can inherit it, and don't need to pass it here